lib/Driver/ToolChains.cpp

```diff
@@ -954,6 +1023,9 @@ toolchains::Darwin::constructInvocation(const LinkJobAction &job,
     Arguments.push_back("-application_extension");
   }
 
+  if (context.OI.SelectedSanitizer == SanitizerKind::Address)
+    AddLinkSanitizerLibArgs(context.Args, Arguments, "asan", *this);
```

Comment:
We are not adding compiler-rt library here. We add the asan library and the code coverage will add its own library.